polaris-react/src/components/Frame/Frame.tsx

```diff
@@ -279,28 +273,67 @@ class FrameInner extends PureComponent<CombinedProps, State> {
 
     return (
       <FrameContext.Provider value={context}>
-        <div
-          className={frameClassName}
-          {...layer.props}
-          {...navigationAttributes}
-        >
-          {skipMarkup}
-          {topBarMarkup}
-          {navigationMarkup}
-          {contextualSaveBarMarkup}
-          {loadingMarkup}
-          {navigationOverlayMarkup}
-          <main
-            className={styles.Main}
-            id={APP_FRAME_MAIN}
-            data-has-global-ribbon={Boolean(globalRibbon)}
-          >
-            <div className={styles.Content}>{children}</div>
-          </main>
-          <ToastManager toastMessages={toastMessages} />
-          {globalRibbonMarkup}
-          <EventListener event="resize" handler={this.handleResize} />
-        </div>
+        <UseFeatures>
+          {(features) => (
+            <div
+              className={classNames(
+                features?.dynamicTopBarAndReframe
```

Comment:
Instead of copying the entire `Frame`, `Main`, and `Content` classes, could we optionally apply class overrides instead?

```
styles.Frame,
styles?.dynamicTopBarAdnReframe && styles['Frame-TopBarAndReframe'],
...
```

That way we can easily remove the or merge only the styles that have changed.